libs/env/context.go

```diff
@@ -63,6 +64,21 @@ func Set(ctx context.Context, key, value string) context.Context {
 	return setMap(ctx, m)
 }
 
+func homeEnvVar() string {
+	if runtime.GOOS == "windows" {
+		return "USERPROFILE"
+	}
+	return "HOME"
+}
+
+func WithUserHomeDir(ctx context.Context, value string) context.Context {
+	return Set(ctx, homeEnvVar(), value)
+}
+
+func UserHomeDir(ctx context.Context) string {
+	return Get(ctx, homeEnvVar())
```

Comment:
So `os.UserHomeDir` returns an error if it is empty. This silently returns an empty string. We should return an error here as well to avoid doing the wrong thing when the home directory inadvertently is not set.

Reply:
which can be the cases when HOME is empty?

Reply:
Users may exec into an environment where it is not set, i.e. a container, or empty env.

The important bit is that we should never silently do the wrong thing, so we should either return an error from this function (reasonable), or panic if the value is empty (not great).